Possible permission errors in blackbox tests when cloning with weird umask

<!-- Thank you for filing an issue to help us improve Dune! -->
## Expected Behavior

On a fresh clone, after `make dev-switch` and `make bootstrap`, `make test` succeeds fully.

## Actual Behavior

Permission errors in the blackbox tests:

```shell
# for context
≻ _boot/dune.exe --version
3.12.0-73-gda55f91
≻ umask
0002
≻ make dune build @remove-write-permissions
./_boot/dune.exe build @remove-write-permissions
File "test/blackbox-tests/test-cases/directory-targets/remove-write-permissions.t", line 1, characters 0-0:
------ test/blackbox-tests/test-cases/directory-targets/remove-write-permissions.t
++++++ test/blackbox-tests/test-cases/directory-targets/remove-write-permissions.t.corrected
File "test/blackbox-tests/test-cases/directory-targets/remove-write-permissions.t", line 17, characters 0-1:
 |Write permissions on directory targets.
 |
 |  $ cat >dune-project <<EOF
 |  > (lang dune 3.4)
 |  > (using directory-targets 0.1)
 |  > EOF
 |
 |  $ cat >dune <<EOF
 |  > (rule
 |  >  (action (system "mkdir -p foo/foo2 && touch foo/foo2/bar"))
 |  >  (targets (dir foo)))
 |  > EOF
 |
 |  $ dune build ./foo
 |  $ dir=_build/default/foo
 |  $ dune_cmd stat permissions $dir
-|  755
+|  775
 |  $ dune_cmd stat permissions $dir/foo2
-|  755
+|  775
 |  $ dune_cmd stat permissions $dir/foo2/bar
 |  444
make: *** [Makefile:200 : dune] Erreur 1



≻ umask 022
≻ make dune build @github1426
./_boot/dune.exe build @github1426
File "test/blackbox-tests/test-cases/github1426.t/run.t", line 1, characters 0-0:
------ test/blackbox-tests/test-cases/github1426.t/run.t
++++++ test/blackbox-tests/test-cases/github1426.t/run.t.corrected
File "test/blackbox-tests/test-cases/github1426.t/run.t", line 3, characters 0-1:
 |Changing write permissions of a dependency doesn't cause a re-run
 |  $ chmod -w ./script.sh
+|  chmod: ./script.sh: new permissions are r-xrwxr-x, not r-xr-xr-x
+|  [1]
 |  $ dune build a_target --display short
 |        script a_target
 |  $ chmod +w ./script.sh
 |  $ dune build a_target --display short
make: *** [Makefile:200 : dune] Erreur 1
```

## Reproduction

Some notes:
- I have no idea why my umask was set at 0002, and not 0022 as it seems to be standard.
- Changing from `umask 0002` to `umask 0022` after cloning the repo doesn't fix the whole of it.
  Changing it before the clone does fix everything however.


## Specifications

- Operating system (distribution and version):
  ubuntu 22-04

--- a/test/blackbox-tests/test-cases/directory-targets/copy-sandboxing.t
+++ b/test/blackbox-tests/test-cases/directory-targets/copy-sandboxing.t
@@ -1,6 +1,8 @@
 Sandbox a rule that depends on a directory target using the copying sandbox
 mode:
 
+  $ umask 022
+
   $ cat >dune-project <<EOF
   > (lang dune 3.8)
   > (using directory-targets 0.1)

--- a/test/blackbox-tests/test-cases/directory-targets/remove-write-permissions.t
+++ b/test/blackbox-tests/test-cases/directory-targets/remove-write-permissions.t
@@ -1,5 +1,7 @@
 Write permissions on directory targets.
 
+  $ umask 022
+
   $ cat >dune-project <<EOF
   > (lang dune 3.4)
   > (using directory-targets 0.1)

